### ISSUE TITLE
Looping in some figures interested in the interoperability of GraphQL provisioned end-points

### ISSUE THREAD

**Issue by loganpowell** [OpenGraphQL/opengraphql-organization#1]
@kevinSuttle started a relevant thread on JSON-LD: https://github.com/graphql/graphql-js/issues/271
@KyleAMathews started another inviting thread here: https://github.com/graphql/graphql-js/issues/490
@nodkz has created an exciting project: https://github.com/nodkz/graphql-compose
I've seen @linonetwo in a number of relevant contexts including references to all of the above.

Gentlemen, @helfer and I are also interested in making open data accessible and interoperable via GraphQL. We would like to invite your thoughts and - if interested - contribution. Let's start with some general thought from you on how we might potentiate your ideas and work!

**Comment by nodkz** [comment]
I have a huge practical experience with reassembling, combining and creating GraphQL schemas. All solutions for problems which I met in our private app with creating complex GraphQL schemas was published via `graphql-compose` and its plugins.

A conceptual model of interoperability which I use:
- `graphql-js` provides definition over configuration (and of course runtime engine)
- `graphql-compose` - is a imperative tool worked on top of `graphql-js`. It provides some methods for creating types and GraphQL Models (so I call types with a list of common resolvers) for further building of complex relations.
- `graphql-compose-*` - is a declarative generators/plugins worked on top of `graphql-compose`, which take some ORMs, schema definitions and creates GraphQL Models from them or modify existed GraphQL Types.

Before `graphql-compose*` was two or three attempts to realize some tool which helps build schemas (all they fail). And for now, I found that `graphql-compose*` concept is very flexible and strong (and tested by time). For now its API quite stable and helps to construct your schemas or custom builders very fast. Eg last builder [graphql-compose-elasticsearch](https://github.com/nodkz/graphql-compose-elasticsearch) was written in two weeks (it's very complicated and did quite much work under the hood).

I'll happy to write some articles about `graphql-compose` but it's too difficult for me. English is not my first language, and for me in current state, needs a huge amount of time to write something understandable. So I hope that someone will do it instead of me. For now, I have not too much free time and some things are waiting for me since last year (eg https://github.com/graphql-compose/graphql-perf).

Honestly, I did not fully understand goals of opengraphql organization (what exactly you want to do). Create some SPEC over GraphQL for better interoperable schema definition? Or take existed/real open data and make it accessible via GraphQL endpoint? Or create new tools or improve existed?
 
Anyway, I open for discussion and collaboration.

**Comment by loganpowell** [comment]
@nodkz that acronym is hilarious :D though I think we'd all agree that - though the transition may not be painless - it's inevitable and a good thing, right?!

To your question on understanding what OpenGraphQL should be about... we're starting to brainstorm about how we can make open datasets interoperable and more transparent. I will just steal from the issues above to give greater insight into our thinking thus far:

As Kevin - in his [request](https://github.com/graphql/graphql-js/issues/271) for first class support of JSON-LD-like 'expansion and compaction':
These features allow users to "merge data from multiple sources." This is one objective of OpenGraphQL, to take the power of GraphQL as a data-format agnostic API layer to create a sort of facade over publicly available APIs. With an OSS/shared community-powered schema, data interoperability is no-longer dependent upon the owners of the data (which I can tell you from experience is not a best bet), but is delegated to the users.

As Lee Byron mentions in the same issue: 
> GraphQL is not about providing publicly accessible structured data on the internet, it's about providing access to data given the needs of a client 

But it could be both! We could create a 'client' that powers open data for open data users.

Kyle asks: 
> ... of what I'm assuming is the coming avalanche of public GraphQL APIs, it'd often be really handy if you could compose all or parts of a public schema with your private one.

I've been working in open data for five years and would attest to the fact that standards adoption efforts are both less common and less energized than efforts to create standards 😆 ... though [automated tools](https://github.com/yarax/swagger-to-graphql) may help us, there's just no replacement for having a curated schema at your disposal. It may seem like a bit of a pain to wrap every open dataset's REST API manually (or with automated tools) to have all the schema in the same place, but I would argue it's the fastest way to get the job done.

@nodkz your work exemplifies what kind of work we're interested in. I.e., speeding the adoption of GraphQL and thereby accelerating our ability to more easily work with and between APIs. Composability of public APIs is within our reach now.

**Comment by loganpowell** [comment]
looping @yarax into the dialog

**Comment by yarax** [comment]
Hi everyone, if you mentioned me here, probably you meant my https://github.com/yarax/swagger-to-graphql project, which is inter alia about converting JSON schema to GraphQL types. 

So the idea great, of course, in order to popularize GraphQL we must make building types and resolvers as fast and easy as possible (now it's not always so)
I could be wrong, but the first idea would be to distinguish resolvers and types.
We should have different adaptors for different type systems (e.g SQL, proto buffers, apache thrift, XSD etc) and easy way to define resolvers to get data over it's schema.

**Comment by loganpowell** [comment]
looping in @JJediny and an old friend @GUI

**Comment by loganpowell** [comment]
@yarax indeed. There are some already maturing:  https://github.com/postgraphql/postgraphql

**Comment by loganpowell** [comment]
looping in @tomitrescak (his project: https://github.com/tomitrescak/apollo-modules)
and @liamcurry (his project: https://github.com/liamcurry/gql)

a couple issues of interest: 
- https://github.com/apollographql/graphql-server/issues/4
- https://github.com/graphql/graphql-js/issues/223

Folks, I apologize for the cold connection here, but I hope you all will find this project interesting. We're only just starting, so please subscribe to this issue if the topic of integrating open data with GraphQL is of interest. At least you should meet others with shared interests ;)
